offline_imu_cam/plugin.cpp

```diff
@@ -63,20 +65,42 @@ class offline_imu_cam : public ILLIXR::threadloop {
 			{bool(sensor_datum.cam0)},
 		}});
 
-
+		assert(errno == 0 && "Errno should not be set before cam0");
 		std::optional<cv::Mat*> cam0 = sensor_datum.cam0
 			? std::make_optional<cv::Mat*>(sensor_datum.cam0.value().load().release())
 			: std::nullopt
 			;
+		RAC_ERRNO_MSG("offline_imu_cam after cam0");
+
+		assert(errno == 0 && "Errno should not be set before cam1");
 		std::optional<cv::Mat*> cam1 = sensor_datum.cam1
 			? std::make_optional<cv::Mat*>(sensor_datum.cam1.value().load().release())
 			: std::nullopt
 			;
-
-		if (cam0 && cam1) {
-			cv::cvtColor(*cam0.value(), *cam0.value(), cv::COLOR_BGR2GRAY);
-			cv::cvtColor(*cam1.value(), *cam1.value(), cv::COLOR_BGR2GRAY);
+		RAC_ERRNO_MSG("offline_imu_cam after cam1");
+
+		assert(errno == 0 && "Errno should not be set before cvtColor");
+		if (cam0.has_value() && cam1.has_value()) {
+		    const int num_ch0 = cam0.value()->channels();
+		    const int num_ch1 = cam1.value()->channels();
+
+		    switch (num_ch0) {
+                case 3:
+                    cv::cvtColor(*cam0.value(), *cam0.value(), cv::COLOR_BGR2GRAY);
+                    break;
+                case 1:
+                default:
+                    break;
+            }
+            switch (num_ch1) {
+                case 3:
+                    cv::cvtColor(*cam1.value(), *cam1.value(), cv::COLOR_BGR2GRAY);
+                case 1:
+                default:
+                    break;
+            }
```

Comment:
IMO unnecessary as `offline_imu_cam` is tightly-coupled to `lazy_load` already. It can be part of the "contract" that lazy loud emits grayscale images. I would rather just `assert` that this contract is met in `offline_imu_cam`.

```
assert(blah->channels() == 1 && "Got a color image where grayscale is expected");
```

Reply:
In previous testing, calling `cv:cvtColor` was an issue because the input was not color. In testing this branch, I had the opposite issue- removing the call was an issue because the input was indeed in color. Did the behavior of `lazy_load` change recently?

Reply:
No, I don't think it changed recently.

Since the component is only used here, the behavior "should" be whatever makes it easier for `offline_imu_cam` and its downstream, `open_vins`. I believe loading grayscale images from `cv::imread` is the right way to go.